### ISSUE TITLE
hyperDB vs. FAISS?

### ISSUE THREAD

**Issue by janzheng** [jdagdelen/hyperDB#9]
In all seriousness, how does this compare to FAISS?

**Comment by yurymalkov** [comment]
Haha. The difference depends on the algorithm, dataset and hardware. 
Ann benchmark had this fabulous numpy implementation (https://github.com/erikbern/ann-benchmarks/blob/main/ann_benchmarks/algorithms/bruteforce.py)

**Comment by janzheng** [comment]
Wow that's an incredible resource. Thanks for sharing! 

I'm switching to HyperDB for the giggles. You can't benchmark hype.

**Comment by jdagdelen** [comment]
In all seriousness, I would totally support someone doing a real benchmark on this. HyperDB is actually a practical library for the use case that many devs have (adding knowledge from <100k paragraphs to an LLM.)

**Comment by karencfisher** [comment]
OK, I came across this looking for the possibility of a light weight, local, server less vector DB. I caught on something was amiss though by the phony link to a supposed paper by Andrej Karpathy (besides, the code there looked like a basic brute force search of the entire vector space -- "this," I though "won't scale"). For a minute (at least) I thought this was all cooked up via ChatGPT,

But damnit, something can be made of this. Why not make it the Dodge coin or Shiba Inu (both took the crypto space by storm despite being initially jokes) of vector DBs and turn it into something useful? Maybe the serverless vector DB equivalent as SQLite is to the RDBMS world? For a small vector document store and query solution that can run offline (thus preserving data privacy) there can be a place.

BTW, if you can read the code for Derridaean distance, it is not Derridaean. A hallmark of Derrida (like other French Post-modernists) is that hardly anyone can understand his stuff. ;)

**Comment by jdagdelen** [comment]
Yeah, this repo is a joke, but it is also totally functional. As for the derridean stuff, we’re adding in quantum algorithms so that should be handled.

**Comment by aismlv** [comment]
Haha stumbled upon this while working on a [project](https://github.com/aismlv/minivn) inspired by similar thinking - I did run a quick [benchmark](https://github.com/aismlv/minivn/tree/main/benchmark) if you are interested, together with some thoughts on when this approach might be a more suitable one

(And to preempt any questions, we are a stealth startup building in public and not accepting any funding at the moment)
